mount.c

```diff
@@ -124,7 +124,7 @@ static void update_mount_flags_and_options(unsigned long *mountflags, char *opts
 
 void mount_entries(const char *root, const struct mount_entry *mounts, size_t nmounts, int no_derandomize)
 {
-	mode_t old_mask = umask(0);
+	mode_t old_mask = umask(0022);
```

Comment:
I'm not sure this is doing what you want.  We are trying to create mountpoints mode 0777.  You are changing that to create them mode 0755.  I don't know why.

Reply:
Misread the man page. I'll drop this part of the change from the diff